### ISSUE TITLE
Nightly releases

### ISSUE THREAD

**Issue by adamreisnz** [atom/atom#18829]
Hi team, I have a couple of questions regarding the nightly releases. First of all, I am aware of the fact that these can be unstable and that I am using them at my own risk and peril.

However, I noticed that the last nightly (11) pretty much broke everything. Pressing CMD-W no longer closed tabs (just moved focus to another tab), search no longer worked, errors kept popping up whenever I switched to another tab, etc. In short, it was completely unusable.

My first question is then, what's the release process of the nightlies? Is there no QA performed on them whatsoever? If a nightly like this with so many issues does get released (first time I've had so many problems with a nightly version by the way), does it get rolled back or will an update with fixes be released sooner than the usual cycle?

Second question is, the releases are called nightlies, but I notice they are only released about once a week. Is this correct? Why not do actual nightly releases every day?

Thanks! Your answers will help me reevaluate if I should keep using Nightlies or switch to the beta channel. I just can't stand looking at the yellow beta channel icon. Purple 🚀

**Comment by Aerijo** [comment]
I don't speak for the Atom team, but AFAIK Nightly is just whatever the master branch holds at the time.

> My first question is then, what's the release process of the nightlies?
> Why not do actual nightly releases every day?

Can't answer this one myself, but I would like to see an automated daily release.

> If a nightly like this with so many issues does get released (first time I've had so many problems with a nightly version by the way), does it get rolled back or will an update with fixes be released sooner than the usual cycle?

If it becomes daily, there would be little point to roll back releases. Don't know about the current process though.

> Is there no QA performed on them whatsoever?

Technically all PRs should pass tests, but the CI seems to be plagued with flakey tests right now (e.g., my PR for adding a markdown document to the RFCs folder doesn't pass). Of course, not all functionality is tested, but ideally regressions would be caught in Nightly / Beta, fixed, and tests added to prevent them happening again.

But no, I doubt QA is performed on Nightly releases. There seems to be little point, especially if they become automated daily. The point of Nightly is to catch bugs before they are released to mainstream; if Nightly itself becomes a mainstream release, it's just pushing the bug checking one step back and we've suddenly got two Beta channels (the release frequency of Beta is a separate issue; I think it can benefit from being more frequent than Stable).

> Your answers will help me reevaluate if I should keep using Nightlies or switch to the beta channel.

I use Beta for when I need to trust the editor to not break. Automatic updates don't work on Linux apparently, so I pull and build from source when I need to test something that hasn't been released yet. My main use of Nightly before this was to test if a bug had been fixed on master before raising it. I definitely would not treat it as a stable editor. Even Beta can have serious bugs in it, but the obvious ones (like the ones you mention) are hopefully caught in Nightly.

On an unrelated note, Atom recently upgraded to Electron 3 (on master), so that may be the cause of the sudden issues (I don't know if it's in Nightly yet). I haven't experienced them myself on a build from master though.

**Comment by adamreisnz** [comment]
Thanks for the insights. Yes I assumed as much, but also thought that master branch should be reasonably stable.

I might switch back to beta for a more stable experience. In the end I had to downgrade to 1.35 to avoid the issues.

**Comment by daviwil** [comment]
The reason why you haven't seen Nightly Releases on a daily basis recently is that we're dealing with some test instability that is preventing the releases from completing.  We're working on getting that resolved now.  Regarding QA, that's the point of doing Nightlies, the people who are willing to use it help us with QA by reporting issues like this :)

We'd definitely like more information on the issues you've been seeing, so if you could provide some more details that'd be helpful.  We're planning on shipping a Beta with the Electron 3 upgrade this week so it'd be nice to get any major issues resolved before then.

Thanks!

**Comment by adamreisnz** [comment]
No worries, the main issues that I encountered consistently, even after restarts were:

1. Pressing CMD-W on an active tab didn't close that tab. It switched focus to the previous tab, but the "closed" tab remained open. Only after pressing the X icon on it with the mouse afterwards caused it to close.

2. After a search in files, clicking on a result opened the relevant file, but didn't highlight/focus on the area of the file with the relevant code. Doing a quick search in the file itself for that same search term didn't yield any results, so it was very hard to locate a specific piece of code, as if the local file search was broken completely.

3. Revealing a file in tree view didn't work.

4. The editorconfig package kept throwing errors every time I switched between a tab, filling the workspace with red error messages (https://github.com/sindresorhus/atom-editorconfig/issues/232)

Those were the most obvious issues, there could have been more that we didn't notice.

**Comment by Arcanemagus** [comment]
> Pressing CMD-W on an active tab didn't close that tab.

This is being tracked here: https://github.com/atom/atom-keymap/issues/237

**Comment by robertrossmann** [comment]
> However, I noticed that the last nightly (11) pretty much broke everything. Pressing CMD-W no longer closed tabs (just moved focus to another tab)

I have experienced this problem as well, and after updating the `editorconfig` package with a fix for the `wrapGuide.updateGuide()` breakage the tabs do close as usual again. 👍

**Comment by rsese** [comment]
@adamreisnz - with that `editorconfig` package update @robertrossmann mentioned, is issue (1)  resolved for you too?  Also, are you still seeing (2) and (3)?  I was unable to reproduce those two on 1.37.0-nightly0 on macOS 10.12.6.

**Comment by adamreisnz** [comment]
I will attempt another update later, I reverted back to a 1.36 version as I had a lot of work that needed to be done and it wasn't possible with the 1.37 release. 

Search is still extremely slow though, I suspect it may no longer be ignoring ignored folders.

**Comment by Arcanemagus** [comment]
Since I believe the original questions have been answered, I'm going to close this issue.

If you find further problems in a nightly release please file a _new_ issue for each problem found, making sure to fill out the template so we can get an idea of when things showed up, thanks!
